superset/models/core.py

```diff
@@ -840,61 +863,130 @@ def inspector(self):
-
-    def all_view_names(self, schema=None, force=False):
-        if not schema:
-            if not self.allow_multi_schema_metadata_fetch:
-                return []
-            views_dict = self.db_engine_spec.fetch_result_sets(
-                self, 'view', force=force)
-            return views_dict.get('', [])
+    @cache_util.memoized_func(
+        key=lambda *args, **kwargs: 'db:{db_id}:schema:None:table_list'.format(
+            db_id=kwargs.get('db_id')),
+        use_tables_cache=True)
+    def all_table_names_in_database(self, db_id=None, cache=False,
+                                    cache_timeout=None, force=False):
+        """
+        Parameters need to be passed as keyword arguments.
+        If cache=True, db_id must be passed in for setting cache key
+        """
+        if not self.allow_multi_schema_metadata_fetch:
+            return []
+        tables_dict = self.db_engine_spec.fetch_result_sets(self, 'table')
+        return tables_dict.get('', [])
+
+    @cache_util.memoized_func(
+        key=lambda *args, **kwargs: 'db:{db_id}:schema:None:view_list'.format(
+            db_id=kwargs.get('db_id')),
+        use_tables_cache=True)
+    def all_view_names_in_database(self, db_id=None, cache=False,
+                                   cache_timeout=None, force=False):
+        """
+        Parameters need to be passed as keyword arguments.
+        If cache=True, db_id must be passed in for setting cache key
+        """
+        if not self.allow_multi_schema_metadata_fetch:
+            return []
+        views_dict = self.db_engine_spec.fetch_result_sets(self, 'view')
+        return views_dict.get('', [])
+
+    @cache_util.memoized_func(
+        key=lambda *args, **kwargs: 'db:{db_id}:schema:{schema}:table_list'.format(
+            db_id=kwargs.get('db_id'), schema=kwargs.get('schema')),
+        use_tables_cache=True)
+    def all_table_names_in_schema(self, schema, db_id=None, cache=False,
+                                  cache_timeout=None, force=False):
+        """
+        Parameters need to be passed as keyword arguments.
+        If cache=True, db_id must be passed in for setting cache key
+
+        For unused parameters, they are referenced in
+        cache_util.memoized_func decorator.
+
+        :param enable_cache: whether cache is enabled for the function
+        :type enable_cache: bool
+        :param cache_timeout: timeout in seconds for the cache
+        :type cache_timeout: int
+        :param force: whether to force refresh the cache
+        :type force: bool
+        :param db_id: database id
+        :type db_id: int
+        :param schema: schema name
+        :type schema: str
+        :return: table list
+        :rtype: list
+        """
+        tables = []
+        try:
+            tables = self.db_engine_spec.get_table_names(
+                inspector=self.inspector, schema=schema)
+        except Exception as e:
+            logging.exception(e)
+            pass
```

Comment:
You don't need `pass` here, you only need it in empty blocks (eg, if line 927 didn't have the logging).